pkg/edgehub/controller_test.go

```diff
@@ -0,0 +1,885 @@
+	certfile.Close()
+
+	// this will create plain text PEM file.
+	keyfile, _ := os.Create("/tmp/kubeedge/testData/edge.key")
+	var pemkey1 = &pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: pkey}
+	pem.Encode(keyfile, pemkey1)
+	keyfile.Close()
+}
+
+//TestNewEdgeHubController() tests whether the EdgeHubController returned is correct or not
+func TestNewEdgeHubController(t *testing.T) {
+	tests := []struct {
+		name             string
+		want             *EdgeHubController
+		controllerConfig config.ControllerConfig
+	}{
+		{"TestNewEdgeHubController: Positive Test Case",
+			&EdgeHubController{
+				config: &config.ControllerConfig{
+					HeartbeatPeroid: 150 * time.Second,
+					RefreshInterval: 15 * time.Minute,
+					AuthInfosPath:   "/var/IEF/secret",
+					PlacementUrl:    "https://10.154.193.32:7444/v1/placement_external/message_queue",
+					ProjectID:       "e632aba927ea4ac2b575ec1603d56f10",
+					NodeId:          "fb4ebb70-2783-42b8-b3ef-63e2fd6d242e",
+				},
+				stopChan:   make(chan struct{}),
+				syncKeeper: make(map[string]chan model.Message),
+			},
+			config.ControllerConfig{
+				HeartbeatPeroid: 150 * time.Second,
+				RefreshInterval: 15 * time.Minute,
+				AuthInfosPath:   "/var/IEF/secret",
+				PlacementUrl:    "https://10.154.193.32:7444/v1/placement_external/message_queue",
+				ProjectID:       "e632aba927ea4ac2b575ec1603d56f10",
+				NodeId:          "fb4ebb70-2783-42b8-b3ef-63e2fd6d242e",
+			},
+		}}
+	for _, tt := range tests {
+		edgeHubConfig := config.GetConfig()
+		edgeHubConfig.CtrConfig = tt.controllerConfig
+		t.Run(tt.name, func(t *testing.T) {
+			got := NewEdgeHubController()
+			if !reflect.DeepEqual(got.context, tt.want.context) {
+				t.Errorf("NewEdgeHubController() Context= %v, want %v", got.context, tt.want.context)
+			}
+			if !reflect.DeepEqual(got.config, tt.want.config) {
+				t.Errorf("NewEdgeHubController() Config = %v, want %v", got.config, tt.want.config)
+			}
+			if !reflect.DeepEqual(got.chClient, tt.want.chClient) {
+				t.Errorf("NewEdgeHubController() chClient = %v, want %v", got.chClient, tt.want.chClient)
+			}
+			if !reflect.DeepEqual(got.keeperLock, tt.want.keeperLock) {
+				t.Errorf("NewEdgeHubController() KeeperLock = %v, want %v", got.keeperLock, tt.want.keeperLock)
+			}
+			if reflect.TypeOf(got.stopChan) != reflect.TypeOf(tt.want.stopChan) {
+				t.Errorf("NewEdgeHubController() StopChan = %v, want %v", got.stopChan, tt.want.stopChan)
+			}
+			if !reflect.DeepEqual(got.syncKeeper, tt.want.syncKeeper) {
+				t.Errorf("NewEdgeHubController() SyncKeeper = %v, want %v", got.syncKeeper, tt.want.syncKeeper)
+			}
+		})
+	}
+}
+
+//TestInitial() tests the procurement of the cloudhub client
+func TestInitial(t *testing.T) {
+	tests := []struct {
+		name             string
+		fields           EdgeHubController
+		webSocketConfig  config.WebSocketConfig
+		controllerConfig config.ControllerConfig
+		ctx              *context.Context
+		wantErr          bool
```

Comment:
wantErr -> wantBool ??

Reply:
Here wantErr variable is used to signify whether it is a positive test case or not. It basically is used to tell the user whether the test case is expecting an error or not. If wantErr is true, it means that an error is expected in the test case(negative test case), if it is false then it signifies that no error is expected. I feel that wantBool might not convey the message properly and might not be easy to understand.